tests/integration/components/polaris-button-test.js

```diff
@@ -1,639 +1,333 @@
-    .hasNoClass(
-      'Polaris-Button--iconOnly',
-      'button with icon - with text, does not have icon-only class'
-    );
-});
+  hooks.beforeEach(function() {
+    this.owner.register('component:svg-jar', MockSvgJarComponent);
+  });
 
-test('it supports a component value for `icon`', function(assert) {
-  this.render(hbs`
-    {{polaris-button
-      text=""
-      icon=(component "polaris-avatar"
-        name="Tom"
-      )
-    }}
-  `);
-
-  let buttonIconComponentSelector = buildNestedSelector(
-    buttonIconWrapperSelector,
-    '.Polaris-Avatar'
-  );
-
-  assert
-    .dom(buttonIconComponentSelector)
-    .exists('component renders inside the icon wrapper');
-  assert
-    .dom(buttonIconSelector)
-    .doesNotExist('icon is not rendered inside the icon wrapper');
-});
+  module('<Button />', function() {
+    module('url', function() {
+      hooks.beforeEach(function() {
+        this.set('mockUrl', 'http://google.com');
+      });
+
+      test('renders a link when present', async function(assert) {
+        await render(hbs`{{polaris-button url=mockUrl}}`);
+        assert.dom('a[data-polaris-unstyled]').exists();
+      });
+
+      test('gets passed into the link', async function(assert) {
+        await render(hbs`{{polaris-button url=mockUrl}}`);
+        assert
+          .dom('a[data-polaris-unstyled]')
+          .hasAttribute('href', this.get('mockUrl'));
+      });
+
+      test('renders a button when not present', async function(assert) {
+        await render(hbs`{{polaris-button}}`);
+        assert.dom('button').exists();
+      });
+    });
 
-test('it supports `disclosure`', function(assert) {
-  this.render(hbs`{{polaris-button text="Click me" disclosure=disclosure}}`);
-
-  let button = find('button.Polaris-Button');
-  let disclosureIcon = find(buttonIconWrapperSelector, button);
-  assert.notOk(
-    disclosureIcon,
-    'button without disclosure - no disclosure icon rendered'
-  );
-
-  this.set('disclosure', true);
-
-  let contentSpans = findAll('span.Polaris-Button__Content > span', button);
-  assert.equal(
-    contentSpans.length,
-    2,
-    'button with disclosure - renders both text and disclosure icon'
-  );
-
-  disclosureIcon = contentSpans[1];
-  assert
-    .dom(disclosureIcon)
-    .hasClass(
-      'Polaris-Button__Icon',
-      'button with disclosure - has disclosure icon after text'
-    );
-  let iconSvg = find('span.Polaris-Icon > svg', disclosureIcon);
-  assert.equal(
-    iconSvg.dataset.iconSource,
-    'polaris/caret-down',
-    'button with disclosure - uses `caret-down` as the SVG'
-  );
-});
+    module('children', function() {
+      test('renders the given children into the button', async function(assert) {
+        this.set('label', 'Click me!');
+        await render(hbs`{{#polaris-button}}{{label}}{{/polaris-button}}`);
+        assert.dom('button').hasText(this.get('label'));
+      });
+
+      test('renders the given children into the link', async function(assert) {
```

Comment:
`children` is really react-world.... 😜 
`block-form` ?

Reply:
🤷‍♂️I think we all know what this means, and matching the React code makes the upgrade process much clearer (as I'm finding out this time around 😬)